TypeScript.py

```diff
@@ -760,33 +767,48 @@ def on_selection_modified(self, view):
     # usually called by Sublime when the buffer is modified
     # not called for undo, redo
     def on_modified(self, view):
-        info = self.getInfo(view)
-        if info:
-            info.modified = True
-            if cli.ST2():
-               info.modCount+=1
-            info.lastModChangeCount = self.change_count(view)
-            self.mod = True
-            (lastCommand, args, rept) = view.command_history(0)
-#            print("modified " + view.file_name() + " command " + lastCommand + " args " + str(args) + " rept " + str(rept))
-            if info.preChangeSent:
-                # change handled in on_text_command
-                info.clientInfo.changeCount = self.change_count(view)
-                info.preChangeSent = False
-            elif lastCommand == "insert":
-                if (not "\n" in args['characters']) and info.prevSel and (len(info.prevSel) == 1) and (info.prevSel[0].empty()) and (not info.clientInfo.pendingChanges):
+        logger.log.debug ("enter on_modified " + str(view.id()))
+        # it is a special view
+        if view.window() and view.window().active_view().id() != view.id():
```

Comment:
Make a function `is_special_view` that takes `view.window()` and `view.id()` as arguments.